clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -345,6 +382,20 @@ public ConfigDef define(String name, Type type, Importance importance, String do
     }
 
     /**
+     * Define a new configuration with deprecated names but no special validation logic, no dependents and no custom recommender
+     * @param name              the name of the config parameter
+     * @param deprecatedNames   the deprecated names of the config parameter
+     * @param type              the type of the config
+     * @param importance        the importance of this config
+     * @param documentation     the documentation string for the config
+     * @return This ConfigDef so you can chain calls
+     */
+    public ConfigDef define(String name, Collection<String> deprecatedNames, Type type, Object defaultValue, Importance importance, String documentation) {
```

Comment:
As you said offline, this is really getting out of hand. Should we introduce a builder of some sort? We can do it in a separate PR, but it would be good to do it before the next release so that we can remove the two overloads that have been introduced.

Reply:
Good point.  I added a builder.